File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-1-freeing-sprite.cpp
/**
 * Usage Example: creating a sprite
 **/

#include "splashkit.h"

int main()
{
    // widow to draw the sprite on
    window start = open_window("create_sprite", 600, 600);

    // bitmap for creating a sprite
    bitmap player = load_bitmap("playerBmp", "protSpriteSheet.png");
    bitmap_set_cell_details(player, 31, 32, 4, 3, 12);

    // creating the player sprite
    sprite player_sprite = create_sprite(player);

    // setting the co-ordinates in refrence to the window
    sprite_set_x(player_sprite, 300);
    sprite_set_y(player_sprite, 300);
    
    // creating game loop
    while (!quit_requested())
    {
        process_events();
        clear_screen(COLOR_BLACK);
        draw_sprite(player_sprite);
        refresh_screen();

        // if up key is typed, the sprite is removed
        if (key_typed(UP_KEY))
        {
            free_sprite(player_sprite);
        }

        update_sprite(player_sprite);
        
    }

    return 0;
}

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-1-freeing-sprite.cs
using System;
using SplashKitSDK;
using static SplashKitSDK.SplashKit;

// Window to draw the sprite on
Window start = OpenWindow("createSprite", 600, 600);

// Bitmap for creating a sprite
Bitmap player = LoadBitmap("playerBitmap", "player.png");
player.SetCellDetails(31, 32, 4, 3, 12); // Set cell details

// Creating the player sprite
Sprite playerSprite = CreateSprite(player);

// Setting the coordinates in reference to the window
playerSprite.X = 300;
playerSprite.Y = 300;

// Game loop
bool spriteExists = true; // Track if the sprite exists
while (!QuitRequested())
{
    ProcessEvents();
    ClearScreen(Color.Black);

    if (spriteExists)
    {
        DrawSprite(playerSprite);
        UpdateSprite(playerSprite);
    }

    RefreshScreen();

    // If UP key is typed, the sprite is removed
    if (spriteExists && KeyTyped(KeyCode.UpKey))
    {
        FreeSprite(playerSprite);
        spriteExists = false; // Set bool to false to stop drawing/updating
    }
}

// Cleanup
if (spriteExists)
{
    FreeSprite(playerSprite);
}

start.Close();